search-sentiments-map/routes/search.js

```diff
@@ -0,0 +1,253 @@
+ * @param {string} countryCode 2 letter country code for search results to be
+ *     written in.
+ * @param {string} query Search query.
+ * @return {Object} Formatted object with country search result data and
+ *     country overall score.
+ */
+async function getSearchResultsForCountryFromAPI(countryCode, query) {
+  let response =
+    await fetch('https://www.googleapis.com/customsearch/v1?key=AIzaSyDszWv1aGP7Q1uOt74CqBpx87KpkhDR6Io&cx=017187910465527070415:o5pur9drtw0&q=' + query + '&cr=' + countryCode + '&num=10&safe=active&dateRestrict=d1&fields=items(title,snippet,htmlTitle,link)');
+  let searchResults = await response.json();
+  return await formatCountryResults(searchResults);
+}
+
+/**
+ * Formats the current results given the results JSON obtained from theAPI.
+ * @param {Object} searchResultsJson Object with information for top 10 search
+ *     results.
+ * @return {Object} Formatted object with country search result data and
+ *     country overall score.
+ */
+async function formatCountryResults(searchResultsJson) {
+  // Parse the JSON string and pass each search result to add to the
+  // countryData object.
+  let currentSearchResults = searchResultsJson.items;
+    let countryData = [];
+    let totalScore = 0;
+    if (currentSearchResults == undefined) {
+      return {score: 0, results: countryData};
+    } else {
+      for (let i = 0; i < currentSearchResults.length; i++) {
+        let formattedResults =
+            await formatSearchResults(currentSearchResults[i]);
+        countryData.push(formattedResults);
+        totalScore += formattedResults.score;
+      }
+      let avgScore = 0;
+      if (currentSearchResults.length !== 0) {
+        avgScore = totalScore / currentSearchResults.length;
+      } 
+      return {score: avgScore, results: countryData};
+    }
+}
+
+/**
+ * Formats search result object.
+ * @param {Object} searchResult Object with information for one search result.
+ * @return {Object} Formatted search result data in JSON form.
+*/
+function formatSearchResults(searchResult) {
+  return getSentiment(searchResult).then(response => response.json())
+    .then((result) => {
+      searchResultData = {
+        title: searchResult.title,
+        snippet: searchResult.snippet,
+        htmlTitle: searchResult.htmlTitle,
+        link: searchResult.link,
+        score: result.score,
+      };
+      return searchResultData;
+    });
+}
+
+/** 
+ * Gets the sentiment score of a search result. 
+ * @param {Object} searchResult Object for one search result.
+ */
+function getSentiment(searchResult) {
+  return fetch('https://trending-search-sentiments.ue.r.appspot.com/sentiment', {
+    method: 'POST',  // Send a request to the URL.
+    headers: new Headers({
+      'Content-Type': 'text/plain',
+    }),
+    body: searchResult.title + searchResult.snippet
+  }).catch(err => {
+    console.log(err);
+  });
+}
+
+/** Deletes search results from 7 days ago. */
```

Comment:
make this comment more generic. say that you delete stale search results and then the threshold you've picked.